S2VX.Game/Story/Command/S2VXCommand.cs

```diff
@@ -0,0 +1,156 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using osu.Framework.Graphics;
+using System;
+using System.Globalization;
+
+namespace S2VX.Game.Story.Command {
+    public abstract class S2VXCommand : IComparable<S2VXCommand> {
+        public abstract CommandType Type { get; set; }
+        public double StartTime { get; set; }
+        public double EndTime { get; set; }
+        public Easing Easing { get; set; } = Easing.None;
+        public abstract void Apply(double time, S2VXStory story);
+
+        public int CompareTo(S2VXCommand other) => StartTime.CompareTo(other.StartTime);
+
+        protected abstract string ToValues();
+        public override string ToString() => $"{Type}|{StartTime}|{EndTime}|{Easing}|{ToValues()}";
+
+        public static S2VXCommand FromString(string data) {
+            var split = data.Split("|");
+            var type = Enum.Parse<CommandType>(split[0]);
+            S2VXCommand command = null;
+            switch (type) {
```

Comment:
Would it be better to have these large switch statements for each command (and add to it for each new command) or to use something like `System.Type.GetType("S2VX.Game.Story.Command." + split[0] + "Command");` and `(S2VXCommand)Activator.CreateInstance(t)`?

Reply:
This would be great. I'll look into it.